pkg/registry/service/strategy.go

```diff
@@ -118,3 +118,22 @@ func MatchServices(label labels.Selector, field fields.Selector) generic.Matcher
 func ServiceToSelectableFields(service *api.Service) fields.Set {
 	return generic.ObjectMetaFieldsSet(service.ObjectMeta, true)
 }
+
+type serviceStatusStrategy struct {
+	svcStrategy
+}
+
+var StatusStrategy = serviceStatusStrategy{Strategy}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update of status
+func (serviceStatusStrategy) PrepareForUpdate(obj, old runtime.Object) {
```

Comment:
please add a TestServiceStatusStrategy in strategy_test that verifies that spec is overwritten